File: solutions/1441. Build an Array With Stack Operations/1441.py
# Time complexity: O(mn)
# Space complexity: O(m)
class Solution:
  def buildArray(self, target: List[int], n: int) -> List[str]:
    ans = []
    i = 0

    for num in target:
      while i < num - 1:
        ans.append("Push")
        ans.append("Pop")
        i += 1

      ans.append("Push")
      i += 1

    return ans

File: solutions/1441. Build an Array With Stack Operations/1441.rs
// Time complexity: O(mn)
// Space complexity: O(m)
impl Solution {
    pub fn build_array(target: Vec<i32>, n: i32) -> Vec<String> {
        let mut ans: Vec<String> = Vec::new();
        let mut i = 0;

        for num in target {
            while i < num - 1 {
                ans.push("Push".to_string());
                ans.push("Pop".to_string());
                i += 1;
            }

            ans.push("Push".to_string());
            i += 1;
        }

        ans
    }
}
